File: benchmarks/r/01_snapshot.R
# Benchmark 01: CDM Snapshot
# R equivalent: CDMConnector::snapshot()

source("benchmarks/r/00_helpers.R")

cat("=== 01: CDM Snapshot ===\n")

t0 <- proc.time()

con <- dbConnect(duckdb(), dbdir = DB_PATH)
cdm <- cdmFromCon(
  con = con,
  cdmSchema = "main",
  writeSchema = "main",
  cdmName = "synthea_1k"
)

snap <- snapshot(cdm)
save_result(snap, "01_snapshot")

elapsed <- (proc.time() - t0)[["elapsed"]]
save_timing("01_snapshot", elapsed)

cdmDisconnect(cdm)
cat("Done in", elapsed, "seconds\n")

File: benchmarks/python/01_snapshot.py
"""Benchmark 01: CDM Snapshot — omopy.connector.snapshot()"""

from helpers import Timer, connect_cdm, save_result, save_timing

from omopy.connector import snapshot

print("=== 01: CDM Snapshot ===")
t = Timer()
cdm = connect_cdm()

snap = snapshot(cdm)
save_result(snap.to_polars(), "01_snapshot")
save_timing("01_snapshot", t.elapsed())
print(f"Done in {t.elapsed():.2f} seconds")
